bcda/api.go

```diff
@@ -162,49 +189,75 @@ func bulkRequest(t string, w http.ResponseWriter, r *http.Request) {
 			return
 		}
 		beneficiaryIds = append(beneficiaryIds, id)
-	}
+		if len(beneficiaryIds) >= maxBeneficiaries {
+
+			args, err := json.Marshal(jobEnqueueArgs{
+				ID:             int(newJob.ID),
+				ACOID:          acoId,
+				UserID:         userId,
+				BeneficiaryIDs: beneficiaryIds,
+				ResourceType:   t,
+				// TODO: remove `Encrypt` when file encryption disable functionality is ready to be deprecated
+				Encrypt: encrypt,
+			})
+			if err != nil {
+				log.Error(err)
+				oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, "", responseutils.Processing)
+				responseutils.WriteError(oo, w, http.StatusInternalServerError)
+				return
+			}
 
-	// TODO: this checks for ?encrypt=false appended to the bulk data request URL
-	// By default, our encryption process is enabled but for now we are giving users the ability to turn
-	// it off
-	// Eventually, we will remove the ability for users to turn it off and it will remain on always
-	var encrypt bool = true
-	param, ok := r.URL.Query()["encrypt"]
-	if ok && strings.ToLower(param[0]) == "false" {
-		encrypt = false
-	}
+			j := &que.Job{
+				Type: "ProcessJob",
+				Args: args,
+			}
 
-	args, err := json.Marshal(jobEnqueueArgs{
-		ID:             int(newJob.ID),
-		ACOID:          acoId,
-		UserID:         userId,
-		BeneficiaryIDs: beneficiaryIds,
-		ResourceType:   t,
-		// TODO: remove `Encrypt` when file encryption disable functionality is ready to be deprecated
-		Encrypt: encrypt,
-	})
-	if err != nil {
-		log.Error(err)
-		oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, "", responseutils.Processing)
-		responseutils.WriteError(oo, w, http.StatusInternalServerError)
-		return
+			if err = qc.Enqueue(j); err != nil {
+				log.Error(err)
+				oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, "", responseutils.Processing)
+				responseutils.WriteError(oo, w, http.StatusInternalServerError)
+				return
+			}
+			jobCount++
+			beneficiaryIds = []string{}
+		}
 	}
+	// Get the last chunk of beneficiaries queued up if there are any.
+	if len(beneficiaryIds) > 0 {
```

Comment:
Looks like a lot of DRY in this conditional block.  How can we alleviate the repeated code?

Reply:
I tried to figure that out and didn't have any good ideas.  It is doing literally 3 things:
Marshall the job args
Que the job
increment the job count
The rest of it is necessary handling for the errors.  I can do something like args, err := dojobMarshall(int(newJob.ID), acoId, userId, beneficiaryIds, t, encrypt) but that hardly seems worth it and a function with that many args makes me sad.

Reply:
What if the first conditional is 
```
if len(beneficiaryIds) >= maxBeneficiaries || len(beneficiaryIds) > 0
   DO STUFF
```

and the second conditional is just 
```
if len(beneficiaryIds) > 0
   DO MORE STUFF
```

Reply:
That isn't it.  I am giving you $123 in 20's.  After I give you 6 twenties I still owe you $3.  That code is to make sure I get you the other $3

Reply:
I think I fixed this.  I just needed a count to know how far along I was in the process